webdictionary/src/test/java/com/eis0/kademlianetwork/IdFinderHandlerTest.java

```diff
@@ -0,0 +1,217 @@
+public class IdFinderHandlerTest {
+    //SEARCHER
+    private SMSPeer SEARCHER_PEER;
+    private KademliaId SEARCHER_ID;
+    private SMSKademliaNode SEARCHER;
+    //NODE1
+    private SMSPeer VALID_PEER1;
+    private KademliaId VALID_NODE1_ID;
+    private SMSKademliaNode VALID_NODE1;
+    //NODE2
+    private SMSPeer VALID_PEER2;
+    private KademliaId VALID_NODE2_ID;
+    private SMSKademliaNode VALID_NODE2;
+
+    private KademliaJoinableNetwork networkMock;
+    private SMSManager smsManagerMock;
+    private SMSKademliaRoutingTable routingTable;
+    private RequestsHandler requestsHandler;
+
+
+    @Before
+    public void setup() {
+        //Searcher
+        SEARCHER_PEER = new SMSPeer("+393423541601");
+        SEARCHER_ID = new KademliaId(SEARCHER_PEER);
+        SEARCHER = new SMSKademliaNode(SEARCHER_PEER);
+        //Node1
+        VALID_PEER1 = new SMSPeer("+393335552121");
+        VALID_NODE1_ID = new KademliaId(VALID_PEER1);
+        VALID_NODE1 = new SMSKademliaNode(VALID_PEER1);
+        //Node2
+        VALID_PEER2 = new SMSPeer("+393423541602");
+        VALID_NODE2_ID = new KademliaId(VALID_PEER2);
+        VALID_NODE2 = new SMSKademliaNode(VALID_PEER2);
+        //RoutingTable
+        routingTable = new SMSKademliaRoutingTable(SEARCHER, new DefaultConfiguration());
+
+        //MOCK
+        networkMock = mock(KademliaJoinableNetwork.class);
+        smsManagerMock = mock(SMSManager.class);
+        requestsHandler = mock(RequestsHandler.class);
+        PowerMockito.mockStatic(SMSManager.class);
+        PowerMockito.when(SMSManager.getInstance()).thenReturn(smsManagerMock);
+
+        PowerMockito.mockStatic(KademliaJoinableNetwork.class);
+        PowerMockito.when(KademliaJoinableNetwork.getInstance()).thenReturn(networkMock);
+
+        when(networkMock.getLocalNode()).thenReturn(SEARCHER);
+        when(networkMock.getLocalRoutingTable()).thenReturn(routingTable);
+
+        when(networkMock.isAlive(SEARCHER_PEER)).thenReturn(true);
+        when(networkMock.isAlive(VALID_PEER1)).thenReturn(true);
+        when(networkMock.isAlive(VALID_PEER2)).thenReturn(false);
+
+
+        when(networkMock.getRequestsHandler()).thenReturn(requestsHandler);
+
+        //when(networkMock.isNodeInNetwork(VALID_NODE2)).thenReturn(true);
+
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void nullId_throws() {
+        IdFinderHandler.searchId(null, SEARCHER_PEER);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void nullSearcher_throws() {
+        IdFinderHandler.searchId(VALID_NODE1_ID, null);
+    }
+
+    /**
+     * This test starts a searchId operation, that should end with the calling Request being closed,
+     * but the Request doesn't exist; as it is written, the code simply ignore the situation after
+     * having started the research
+     */
+    @Test
+    public void nullRequest_throws() {
+        IdFinderHandler.searchId(VALID_NODE1_ID, SEARCHER_PEER);
+    }
```

Comment:
I don't understand what you're testing here.
In the comment you say the request is ignored, so it's executed until the end where nothing happens, but the test is called `nullRequest_throws` but you're not checking for any exception (`@Test`)

Reply:
In fact, there is no exception to catch because of the structure of the code this searchId operation is not bound to any Request (it has not been originated by a FindIdRequest), but as you can see in the sendResult() method if the local node is the searching node the end of the research cycle is a call to the completeFindIdRequest() method.
As you remember (considering that you wrote it) the completeFindIdRequest() method checks if the Request exists, if it doesn't there is no Exception and the cycle simply ends. So I created a simple test whose purpose was just to try to complete a searchId(), and the lack of Exceptions was already a success; I have to admit that it became redundant after the other tests

Reply:
Then the only thing to do here, is to change the name of the test to a more fitting name